packages/viz/src/lib/Layer.ts

```diff
@@ -115,24 +108,35 @@ export class Layer {
     return this._deckLayer;
   }
 
+  private async _getLayerProperties() {
+    const metadata = this._source.getMetadata();
+
+    const styleProps = this._style
+      ? this._style.getProperties(this._source)
+      : undefined;
```

Comment:
This will break below usage of `styleProps` if style is undefined. We need to use an empty object so that it doesn't break.

And as far as I see, style will never be a falsy value. It is turned into a class in the constructor and whenever we replace it, so I think it's safe to use `this._style.getProperties(this._source)` below. What do you think?

Reply:
yup. You're right! it will never be falsy